ml/src/main/scala/frameless/ml/internals/SelectorByValue.scala

```diff
@@ -0,0 +1,29 @@
+package frameless
+package ml
+package internals
+
+import shapeless.labelled.FieldType
+import shapeless.{::, DepFn1, HList, Witness}
+
+/**
+  * Typeclass supporting record selection by value type (returning the first key whose value is of type `Value`)
+  */
+trait SelectorByValue[L <: HList, Value] extends DepFn1[L] with Serializable { type Out <: Symbol }
```

Comment:
I think this could be up-streamed to shapeless

Reply:
I think you use it at the type level, you might to remove the apply altogether for this usage.

Reply:
Yes, upstreaming this to shapeless as my intention :) (that's why I also added the apply pattern). But I prefer to include it in this PR to avoid having to wait for a new version of shapeless to be published. Once I've did the shapeless PR and it is merged, I'll remove it from there.